servicesExternal/ipfs_service.py

```diff
@@ -23,21 +23,33 @@ def __init__(self, publicIpfs: bool=True):
         self.ipfs = ipfsapi.connect(self.ipfsConfig["host"], self.ipfsConfig["port"], self.ipfsConfig["protocol"])
 
     async def add_file(self, data: str):
-        if data == None or data == "":
-            raise ValueError("")
+        if data is None or data == "":
+            raise ValueError("Data is None or invalid.")
         
         data_parsed = json.loads(data)
         result = await self.ipfs.add(json.dumps(data_parsed, separators=(',',':')))
 
         return result["hash"]
 
     async def get_file(self, hash: str):
-        if hash == None or hash == "":
-            raise ValueError("")
+        if hash is None or hash == "":
+            raise ValueError("Hash is None or invalid.")
         
         try:
             result = await self.ipfs.cat(hash)
 
             return result
+        except ipfsapi.exceptions.ConnectionError as err:
+            print("IPFS connection error: ", err)
+        except ipfsapi.exceptions.ProtocolError as err:
+            print("Bad connection protocol: ", err)
+        except ipfsapi.exceptions.ErrorResponse as err:
+            print("Error retrieving file: ", err)
+        except ipfsapi.exceptions.StatusError as err:
+            print("Status error: ", err)
+        except ipfsapi.exceptions.TimeoutError as err:
+            print("Connection timeout error: ", err)
         except ipfsapi.exceptions.Error as err:
-            raise err
\ No newline at end of file
+            print("Error retrieving file: ", err)
+        finally:
+            return None
```

Comment:
Looking at original project and file I am not really sure what to return here. Should None be sufficient?

Reply:
Once the `except`s are changed to `raise` instead of `print` you can get rid of the `finally` and `None` will be default returned by functions that don't return anything.